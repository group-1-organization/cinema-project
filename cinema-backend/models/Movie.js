const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const movieSchema = new Schema({
    title: {
        type: String,
        minlength: 2,
        required: true
    },
    description: {
        type: String,
        minlength: 2,
        required: true
    },
    poster: {
        type: String,
        minlength: 2,
        required: true
    },
    actors: {
        type: Array,
        minlength: 1,
        required: true
    },
    director: {
        type: String,
        minlength: 2,
        required: true
    },
    showings: {
        type: Array,
        minlength: 1,
        required: true
    }
});

module.exports = mongoose.model("Movies", movieSchema);