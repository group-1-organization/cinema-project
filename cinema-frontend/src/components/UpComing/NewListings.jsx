import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import NewMovie from './NewMovie';
import axios from 'axios';
import { useState } from 'react';
import { useEffect } from 'react';
import '../../App.css';


const NewListings = () => {
    const [movies, setMovies] = useState([]);

    useEffect(() => {
        getMovies();
    }, []);


    const getMovies = async () => {
        await axios.get("http://localhost:5000/cinema/newmovies").then((response) => {
            setMovies(response.data);
        }).catch((error) => {
            console.log(error);
        })
    };

    console.log(movies)

    return (
        <div>
            <div className="page-header">
                <h1>Listings page</h1>
            </div>
            <br />
            <Container>


                {/* <Row className="border-bottom"></Row> */}
                {movies.map((movie) => (
                    <NewMovie movie={movie} />
                ))}
            </Container>
        </div>
    )
}

export default NewListings
