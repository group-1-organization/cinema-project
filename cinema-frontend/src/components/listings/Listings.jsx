import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import axios from 'axios';
import { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';

const Listings = () => {

    const [movies, setMovies] = useState([]);

    useEffect(() => {
        getMovies();
    }, []);

    const getMovies = async () => {
        await axios.get("http://localhost:5000/cinema/movies").then((response) => {
            setMovies(response.data);
        }).catch((error) => {
            console.log(error);
        })
    };

    return (
        <Container fluid>
            <h1>Listings page</h1>

            <Row>
                {movies.map((movie) => (
                    <Col sm={3}>
                        <Link to={`/movie/${movie.title}`}><img src={movie.poster} width="300rem" /></Link>
                    </Col>
                ))}
            </Row>
        </Container>
    )
}
export default Listings
