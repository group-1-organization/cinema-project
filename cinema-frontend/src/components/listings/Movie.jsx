import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Image from 'react-bootstrap/Image';
import Button from 'react-bootstrap/Button';
import Container from 'react-bootstrap/Container';
import { useParams } from 'react-router-dom';
import { useState, useEffect } from 'react';
import axios from 'axios';

const Movie = () => {

    const { title } = useParams();
    const [movie, setMovie] = useState({});
    const [loaded, setLoaded] = useState(false);
    const [error, setError] = useState(null);
    const [actors, setActors] = useState(["a1", "a2"]);
    const [showings, setShowings] = useState(["a1", "a2"]);
    const [poster, setPoster] = useState("");

    useEffect(() => {
        getMovie();
    }, []);

    const getMovie = async () => {
        await axios.get(`http://localhost:5000/cinema/movies/name/${title}`).then((response) => {
            setLoaded(true);
            setActors(response.data[0].actors);
            setShowings(response.data[0].showings);
            setPoster(response.data[0].poster);
            setMovie(response.data[0]);
        }).catch((error) => {
            setLoaded(true);
            setError(error);
            console.log(error);
        })
    };
    if (error) {
        return (
            <p>error</p>
        );
    } else if (!loaded) {
        return (<p>loading...</p>);
    } else {
        return (
            <Container>
                <Row key={movie._id} className="border-bottom py-3">
                    <img src={`/${movie.poster}`} width="400" height="500" alt={movie.poster} />
                </Row>
                <Row>
                    <Col sm={9} md={10}>
                        <h3>{movie.title}</h3>
                        <p>
                            Actors : {actors.map((actor, index) => (
                                <span key={index}> {actor} | </span>
                            ))}
                        </p>
                        <p>
                            Director : {movie.director}
                        </p>
                        <div>
                            {showings.map((showing) => (
                                <Button className="mr-3" variant="primary">{showing}</Button>
                            ))}
                        </div>
                    </Col>
                </Row>
            </Container>

        )
    }
}

export default Movie
