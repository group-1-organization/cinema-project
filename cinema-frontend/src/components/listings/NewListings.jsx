import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import axios from 'axios';
import { useState } from 'react';
import { useEffect } from 'react';
import '../../App.css';
import { Link } from 'react-router-dom';
import Image from 'react-bootstrap/Image';
import Col from 'react-bootstrap/Col';

const NewListings = () => {

    const [movies, setMovies] = useState([]);

    useEffect(() => {
        getMovies();
    }, []);

    const getMovies = async () => {
        await axios.get("http://localhost:5000/cinema/newmovies").then((response) => {
            setMovies(response.data);
        }).catch((error) => {
            console.log(error);
        })
    };

    console.log(movies)

    return (
        <div>
            <div className="page-header">
                <h1>Listings page</h1>
            </div>
            <br />

            <Container>


                <Row xs={1} sm={3} lg={4}>
                    {movies.map((movie, i) => (
                        <Col key={i} className="mt-4">
                            <Link to={`/movie/${movie.title}`}>
                                <Image src={`/images/${movie.poster}`} fluid />
                            </Link>
                        </Col>
                    ))}
                </Row>
            </Container>


        </div>
    )
}

export default NewListings
