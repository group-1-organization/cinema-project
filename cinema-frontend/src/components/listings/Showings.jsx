import Link from "react-router-dom/Link";
import Button from "react-bootstrap/Button";


const Showings = ({ showing, movie, setTime, setTitle }) => {


    const linkbooking = () => {

        let time = showing;
        let title = movie.title;

        setTime(time);
        setTitle(title);
    }


    return (
        <Link to="/bookings" ><Button className="mr-3" variant="primary" onClick={linkbooking}>{showing} </Button></Link>
    );
}

export default Showings;