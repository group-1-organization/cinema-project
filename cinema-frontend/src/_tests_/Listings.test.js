import renderer, { act } from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Listings from '../components/listings/Listings';
import NewListings from '../components/listings/NewListings';
import Movie from '../components/listings/Movie';
import Showings from '../components/listings/Showings';

test('Listings page matches snapshot', () => {
    const tree = renderer.create(<Listings />);

    expect(tree).toMatchSnapshot();
});

test('New Listings page matches snapshot', () => {
    const tree = renderer.create(<NewListings />);

    expect(tree).toMatchSnapshot();
});

// test('Movie page matches snapshot', () => {
//     const tree = renderer.create(<Movie />);

//     expect(tree).toMatchSnapshot();
// });

test('Showings matches snapshot', () => {
    const tree = renderer.create(<Router><Showings /></Router>);

    expect(tree).toMatchSnapshot();
});