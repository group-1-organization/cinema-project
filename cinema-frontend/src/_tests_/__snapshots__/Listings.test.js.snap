// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listings page matches snapshot 1`] = `
<div>
  <div
    className="page-header"
  >
    <h1>
      Listings page
    </h1>
  </div>
  <br />
  <div
    className="container"
  >
    <div
      className="row row-cols-lg-4 row-cols-sm-3 row-cols-1"
    />
  </div>
</div>
`;

exports[`Movie page matches snapshot 1`] = `
<p>
  loading...
</p>
`;

exports[`New Listings page matches snapshot 1`] = `
<div>
  <div
    className="page-header"
  >
    <h1>
      Listings page
    </h1>
  </div>
  <br />
  <div
    className="container"
  >
    <div
      className="row row-cols-lg-4 row-cols-sm-3 row-cols-1"
    />
  </div>
</div>
`;

exports[`Showings matches snapshot 1`] = `
<a
  href="/bookings"
  onClick={[Function]}
>
  <button
    className="mr-3 button-colour btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
     
  </button>
</a>
`;
